server/src/main/java/org/opensearch/search/ResourceType.java

```diff
@@ -8,12 +8,18 @@
 
 package org.opensearch.search;
 
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
 /**
  * Enum to hold the resource type
  */
+@PublicApi(since = "1.3")
```

Comment:
This class feels like it should be in a common package instead of search?  perhaps `org.opensearch.monitor` or `.common`?  Also this annotation should be 2.x as it was introduced with https://github.com/opensearch-project/OpenSearch/pull/13474/files

Reply:
I was thinking since this feature(SBP) was introduced in 1.3 that is why I had marked it as 1.3. But I will make this change